### PR TITLE
feat(donation): progress bar for recurring goals #195

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix https://github.com/WordImpress/Give-Recurring-Donations/issues/195

## How Has This Been Tested?
Manually tested by enabling the recurring, currency switcher and Fee recovery add-on 

## Task
- [x] Added action after Goal Format radio button 
- [x] Added filter to alter Goal donation Amount, Goal target, Number of Donations, Number of Donors
- [x] Remove duplicate code

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/give-metabox-functions.php [hunk 1]
```diff
@@ -1694,3 +1694,65 @@ function give_repeater_field_set_editor_id( $field_name, $field ) {
...
+	$field['style']         = isset( $field['style'] ) ? $field['style'] : '';
+	$field['wrapper_class'] = isset( $field['wrapper_class'] ) ? $field['wrapper_class'] : '';
+	$field['value']         = give_get_field_value( $field, $thepostid );
+	$field['name']          = isset( $field['name'] ) ? $field['name'] : $field['id'];
+
+	echo '<fieldset class="give-field-wrap ' . esc_attr( $field['id'] ) . '_field ' . esc_attr( $field['wrapper_class'] ) . '"><span class="give-field-label">' . wp_kses_post( $field['name'] ) . '</span><legend class="screen-reader-text">' . wp_kses_post( $field['name'] ) . '</legend><ul class="give-radios">';
+
+	foreach ( $field['options'] as $key => $value ) {
+
+		echo '<li><label><input
```
[the remark this answers is not in the document]
Comment (raftaar1191) [reply includes/admin/give-metabox-functions.php]: Update the PR @ravinderk

File: includes/misc-functions.php [hunk 2]
```diff
@@ -1928,4 +1928,118 @@ function __give_get_active_by_user_meta( $banner_addon_name ) {
...
+		$total_goal = give_currency_filter( give_format_amount( $total_goal ) );
+	}
+
+	$stats_array = array_merge(
+		array(
+			'progress'       => $progress,
+			'actual'         => $actual,
+			'goal'           => $total_goal,
+			'format'         => $goal_format,
+			'goal_donations' => $donations,
```
Comment (ravinderk) [inline includes/misc-functions.php]: @raftaar1191 why we need these extra goal values?

Comment (raftaar1191) [reply includes/misc-functions.php]: In any case, if admin wants to use the filter `give_goal_progress_stats` so s/he does not have fire a query again to get the value of `goal_donations`, `goal_income` and `goal_donors` s/he can easily get it from the filter value

Comment (ravinderk) [reply includes/misc-functions.php]: @raftaar1191  Please remove them and update fn to only run filters when necessary. for example, if goal set to  `donor` then income and donation has no use and function which is attached to these filter will run with even if they do not need to run

Comment (raftaar1191) [reply includes/misc-functions.php]: PR updated as suggested by you @ravinderk and also tested the PR

**Comment by DevinWalker** [comment]
@raftaar1191 please fix the merge conflict

**Comment by raftaar1191** [comment]
@DevinWalker PR updated and conflicts solved

File: includes/admin/give-metabox-functions.php [hunk 3]
```diff
@@ -1694,3 +1694,84 @@ function give_repeater_field_set_editor_id( $field_name, $field ) {
...
+	 *
+	 * @since 2.1.3
+	 *
+	 * @param array $field Array of radio field arguments
+	 */
+	do_action( 'give_donation_form_goal_before_description', $field );
+
+	echo give_get_field_description( $field );
+
+	printf( '</fieldset>' );
```
Comment (DevinWalker) [inline includes/admin/give-metabox-functions.php]: Why `printf` a single end tag?

File: includes/admin/give-metabox-functions.php [hunk 4]
```diff
@@ -1694,3 +1694,84 @@ function give_repeater_field_set_editor_id( $field_name, $field ) {
...
+				give_get_custom_attributes( $field ),
+				esc_html( $value )
+			);
+		}
+		?>
+    </ul>
+
+	<?php
+	/**
+	 * Action to add HTML after donation form radio button is display and before description
```
Comment (DevinWalker) [inline includes/admin/give-metabox-functions.php]: Missing period at end of sentence breaks WPCS rules. Please @raftaar1191 ensure you're using wpcs so you don't make simple mistakes.

**Comment by raftaar1191** [comment]
@DevinWalker updated the PR as suggested buy you

**Pull request merged**